File: server/src/main/java/com/logiflow/server/services/manager/ManagerService.java
package com.logiflow.server.services.manager;

import com.logiflow.server.dtos.manager.ManagerDtos;

import java.time.LocalDate;
import java.util.List;

public interface ManagerService {
    // 1 Dashboard overview
    ManagerDtos.ManagerOverviewDto getDashboardOverview(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 2: Overall operations performance
    ManagerDtos.OperationsPerformanceDto getOperationsPerformance(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 3: Fleet status
    ManagerDtos.FleetStatusDto getFleetStatus();

    // API 4: Deliveries report (summary by day)
    List<ManagerDtos.DeliveryReportItemDto> getDeliveriesReport(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 5: Compliance / Issues Report
    ManagerDtos.IssuesReportResponseDto getIssueReports(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 6: Compliance check
    ManagerDtos.ComplianceCheckResponseDto getComplianceCheck(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 7: Route analytics / summary
    List<ManagerDtos.RouteSummaryItemDto> getRouteSummary(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 8: Alerts
    List<ManagerDtos.AlertDto> getAlerts(
            LocalDate startDate,
            LocalDate endDate
    );

    // API 9: Manager activities / audit log
    List<ManagerDtos.ManagerActivityDto> getManagerActivities(
            LocalDate startDate,
            LocalDate endDate
    );

    List<ManagerDtos.RecommendationDto> getRecommendations(
            LocalDate startDate,
            LocalDate endDate
    );
}

File: client/src/services/manager/managerService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/manager';

// hàm chung lấy header Authorization
function getAuthHeaders() {
    const token = localStorage.getItem('token');
    if (!token) {
        return {};
    }
    return {
        Authorization: `Bearer ${token}`,
    };
}

export async function getOperationsPerformance(startDate, endDate) {
    const response = await axios.get(
        `${API_BASE_URL}/operations/performance`,
        {
            params: { startDate, endDate },
            headers: getAuthHeaders(),
            withCredentials: false,
        }
    );
    return response.data;
}

export async function getFleetStatus() {
    const response = await axios.get(
        `${API_BASE_URL}/fleet/status`,
        {
            headers: getAuthHeaders(),
            withCredentials: false,
        }
    );
    return response.data;
}

export async function getDeliveryReport(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/reports/deliveries`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}

export async function getIssueReports(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/reports/issues`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}

export async function getComplianceCheck(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/compliance/check`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}

export async function getRouteSummary(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/analytics/route-summary`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}

export async function getAlerts(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/alerts`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}

export async function getManagerActivities(startDate, endDate) {
    const response = await axios.get(`${API_BASE_URL}/audit/activities`, {
        params: { startDate, endDate },
        headers: getAuthHeaders(),
        withCredentials: false,
    });
    return response.data;
}
